plugins/tiledb/io/TileDBReader.cpp

```diff
@@ -0,0 +1,298 @@
+/******************************************************************************
+* Copyright (c) 2019 TileDB, Inc
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include <algorithm>
+
+#include <boost/any.hpp>
+
+#include "TileDBReader.hpp"
+
+namespace pdal {
+
+static PluginInfo const s_info
+{
+        "readers.tiledb",
+        "TileDB Reader",
+        "http://pdal.io/stages/readers.tiledb.html"
+};
+
+CREATE_SHARED_STAGE(TileDBReader, s_info)
+std::string TileDBReader::getName() const { return s_info.name; }
+
+template <typename T>
+void makeBuffer(tiledb::Query& query, std::map<std::string, pdalboost::any>& mp,  const std::string& key,
+        const unsigned bufSize)
+{
+    auto sp = std::make_shared<std::vector<T>>(bufSize);
+    mp[key] = sp;
+
+    if (key != TILEDB_COORDS)
+        query.set_buffer(key, *sp);
+}
+
+void TileDBReader::addArgs(ProgramArgs& args)
+{
+    args.add("array_name", "TileDB array name", m_arrayName).setPositional();
+    args.add("config_file", "TileDB configuration file location", m_cfgFileName);
+    args.add("chunk_size", "TileDB read chunk size", m_chunkSize, point_count_t(1000));
```

Comment:
Discussed extensions:
* Subarray support (optional)
* Stats output for benchmarking (optional)

Reply:
:+1: added both